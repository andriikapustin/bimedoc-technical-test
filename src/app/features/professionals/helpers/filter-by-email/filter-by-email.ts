import { IProfessional } from '../../../../models/professional.model';

const regExp = /thomas/gi;

export function filterByEmailAndSecureEmail(professionals: IProfessional[]): IProfessional[] {
  return professionals.filter(
    ({ email, secure_email }) => email?.match(regExp) || secure_email?.some((secureEmail) => secureEmail.match(regExp)),
  );
}
