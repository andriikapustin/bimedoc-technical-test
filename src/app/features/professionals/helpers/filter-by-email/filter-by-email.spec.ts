import { IProfessional } from '../../../../models/professional.model';

import { filterByEmailAndSecureEmail } from './filter-by-email';

describe(`filterByEmailAndSecureEmail()`, () => {
  const professionalsMock = [
    { email: 'Thomas@gmail.com' },
    { secure_email: ['THOMAS@gmail.com'] },
    { email: 'andrii@gmail.com' },
    { email: 'andrii@gmail.com', secure_email: ['thomas@gmail.com'] },
  ] as IProfessional[];

  const expectedMock = [
    { email: 'Thomas@gmail.com' },
    { secure_email: ['THOMAS@gmail.com'] },
    { email: 'andrii@gmail.com', secure_email: ['thomas@gmail.com'] },
  ] as IProfessional[];

  it('should return array', () => {
    expect(filterByEmailAndSecureEmail(professionalsMock)).toBeInstanceOf(Array);
  });

  it('should return correctly filtered array', () => {
    expect(filterByEmailAndSecureEmail(professionalsMock)).toEqual(expectedMock);
  });

  it('should match snapshot', () => {
    expect(filterByEmailAndSecureEmail(professionalsMock)).toMatchSnapshot();
  });
});
