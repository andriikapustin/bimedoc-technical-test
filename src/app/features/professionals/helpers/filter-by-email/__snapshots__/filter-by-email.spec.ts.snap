// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`filterByEmailAndSecureEmail() should match snapshot 1`] = `
Array [
  Object {
    "email": "Thomas@gmail.com",
  },
  Object {
    "secure_email": Array [
      "THOMAS@gmail.com",
    ],
  },
  Object {
    "email": "andrii@gmail.com",
    "secure_email": Array [
      "thomas@gmail.com",
    ],
  },
]
`;
